src/MacroTools/Powers/Transfiguration.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using Microsoft.VisualBasic;
+using WCSharp.Events;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.Powers
+{
+  /// <summary>
+  /// Causes dying units to Transfiguring at a specific location.
+  /// </summary>
+  public sealed class Transfiguration : Power
+  {
+    private readonly float _chance;
+    private readonly Point _returnPoint;
+    private readonly Rectangle _noReturnRect;
+    public int AnimatedArmorID { get; init; } = FourCC("o01I");
```

Comment:
```suggestion
    public required int AnimatedArmorID { get; init; }
```
Map-specific stuff like this should be defined when the Power is constructed, and shouldn't have a default. So you'll need to move the `o01I` bit to wherever this Power is actually given. Also use `Constants` once it's there; the reason you can't use it here is because `MacroTools` doesn't have access to it.

Reply:
Il try my best, but I am not sure how to do that correctly

Reply:
Once you change the above, your IDE will give you an error in the exact place you need to add it. You should then be able to go to that error and press Alt+Enter, take the first recommendation Visual Studio gives you, and it'll mostly set it up for you. Then you just need to type the unit ID in. Let me know if you have any difficulty with that.